oauth-rest/keycloak-custom-providers/src/main/java/com/baeldung/auth/provider/user/CustomUser.java

```diff
@@ -121,4 +122,9 @@ CustomUser build() {
             
         }
     }
+
+    @Override
+    public SubjectCredentialManager credentialManager() {
+        return new LegacyUserCredentialManager(session, realm, this);
```

Comment:
can we use other implementation of SubjectCredentialManager

Reply:
SubjectCredentialManager provides two implementations LegacyUserCredentialManager and MapUserCredentialManager. 
LegacyUserCredentialManager uses UserStorageProvider and MapUserCredentialManager uses MapUserEntity.
Our code is built around UserStorageProvider implementation, hence I have used LegacyUserCredentialManager.